src/python/pants/backend/python/goals/package_pex_binary.py

```diff
@@ -126,9 +128,7 @@ async def package_pex_binary(
         PexFromTargetsRequest(
             addresses=[field_set.address],
             internal_only=False,
-            # TODO(John Sirois): Support ConsoleScript in PexBinary targets:
-            #  https://github.com/pantsbuild/pants/issues/11619
-            main=resolved_entry_point.val,
+            main=resolved_entry_point.val or field_set.script.value,
```

Comment:
Although we would ideally validate this eagerly via #9561, in the meantime it does seem useful to validate it lazily at least?

Reply:
I'm hoping to do #9561 really soon in time for Pants 2.8.0 - possibly as a small weekend hack project. I don't think it will very difficult